File: TrifoliaFhir/Models/PersonListModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TrifoliaFhir.Models
{
    public class PersonListModel
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("name")]
        public Hl7.Fhir.Model.HumanName Name { get; set; }

        [JsonProperty("email")]
        public Hl7.Fhir.Model.ContactPoint Email { get; set; }

        [JsonProperty("organization")]
        public Hl7.Fhir.Model.ResourceReference Organization { get; set; }
    }
}

File: src/app/models/person-list-model.ts
import { HumanName } from './fhir';

export class PersonListModel {
    public name?: HumanName[] = [];

    constructor(obj?: any) {
        if (obj) {
            for (let name of obj.name) {
                const newName = new HumanName(name);
                this.name.push(newName);
            }
        }
    }

    public getDisplayName(): string {
        if (this.name.length > 0) {
            return this.name[0].getDisplay();
        }
    }
}
